### PR TITLE
Handle errors for Github GraphQL API

### PR DESCRIPTION
**Pull request by szheng2207**

### CODE REVIEW

File: apps/github/client/gql_api.go [hunk 1]
```diff
@@ -20,8 +21,22 @@ type Node[Type any] struct {
...
+// Error is GitHub GraphQL API error format:
+// https://github.com/graphql/graphql-spec/blob/main/spec/Section%207%20--%20Response.md#errors
+// where entry "extensions" is GitHub GraphQL API specific error extension attributes
 type Error struct {
-	// TODO(szheng2207): explore Github GraphQL API error structure
+	Message   string `json:"message"`
+	Locations []struct {
+		Line   int `json:"line"`
+		Column int `json:"column"`
+	} `json:"locations"`
```
Comment (naturetime10) [inline apps/github/client/gql_api.go]: Let's extract 
```
struct {
		Line   int `json:"line"`
		Column int `json:"column"`
}
```
into a struct called `Location`?

File: apps/github/client/gql_api.go [hunk 2]
```diff
@@ -20,8 +21,22 @@ type Node[Type any] struct {
...
+	Extensions struct {
+		Value    interface{} `json:"value"`
+		Problems []struct {
+			Path        []interface{} `json:"path"`
+			Explanation string        `json:"explanation"`
+		} `json:"problems"`
```
Comment (naturetime10) [inline apps/github/client/gql_api.go]: Can we extract
```
struct {
			Path        []interface{} `json:"path"`
			Explanation string        `json:"explanation"`
		}
```
into a struct called `Problem`?

File: apps/github/client/gql_api.go [hunk 3]
```diff
@@ -70,10 +85,14 @@ func (g *GraphQLAPI) GetPullRequestByNodeID(ct context.Context, installation Ins
...
+		var multiErr *multierror.Error
+		for _, resErr := range res.Errors {
+			multiErr = multierror.Append(multiErr, errors.New(resErr.Message))
+		}
+
 		internalErr := &errs.Error{
 			Code:     errs.Unknown,
-			EmbedErr: fmt.Errorf("%+v", res.Errors[0]),
+			EmbedErr: multiErr,
 		}
```
Comment (naturetime10) [inline apps/github/client/gql_api.go]: Can we create multiple `internalErr` and log them individually?

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: apps/github/client/gql_api.go (hunk 3 above)
Comment (szheng2207) [reply apps/github/client/gql_api.go]: sure, mind sharing why can't we combine the errors in the log?

Comment (naturetime10) [reply apps/github/client/gql_api.go]: @szheng2207 We can. Let's create our own `MultiErrors` helper.

Comment (naturetime10) [reply apps/github/client/gql_api.go]: @szheng2207 In the next PR.

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**